src/hooks/realtime/use-realtime.ts

```diff
@@ -31,17 +31,22 @@ export type UseRealtimeCompareFn<Data = any> = (
 type CompareFnDefaultData<Data> = Data & { id: any }
 
 export function useRealtime<Data = any>(
-    table: string,
+    options: string | ({ table: string } & UseSelectConfig<Data>),
     compareFn: UseRealtimeCompareFn<Data> = (a, b) =>
         (<CompareFnDefaultData<Data>>a).id ===
         (<CompareFnDefaultData<Data>>b).id,
 ): UseRealtimeResponse<Data> {
```

Comment:
Can we keep the `table` as a string and add a new `config` parameter?

```
export type UseRealtimeConfig<Data> = {
    select?: Omit<UseSelectConfig<Data>, 'pause'>
}

export function useRealtime<Data = any>(
    table: string,
    config: UseRealtimeConfig<Data>,
    compareFn: UseRealtimeCompareFn<Data> = (a, b) =>
        (<CompareFnDefaultData<Data>>a).id ===
        (<CompareFnDefaultData<Data>>b).id,
): UseRealtimeResponse<Data>
```

I appreciate wanting to keep things backward compatible, but it's early enough in the project that I'm fine introducing a breaking change for the sake of a better API foundation for the future.

Reply:
@tmm Code is updated

Indeed this update makes the arguments and the fn api much more explicit and open for extension :rocket: